### ISSUE TITLE
Model without error, returning true in save() method but not saving in database.

### ISSUE THREAD

**Issue by ericmaicon** [yiisoft/yii2#8322]
Hello guys, 

check this peace of code:
```php
$stepSms = new MSmsStep();
$stepSms->smsId = 1;
$stepSms->order = 1;
$stepSms->touch('date');
$saved = $stepSms->save();
```
This piece of code was supposed to work fine, but it doesn't. I checked the $saved var, it returned true. The $stepSms->errors (after a validate() method) and it was an empty array.

I have seen these guys with the same error:
http://stackoverflow.com/questions/8722354/yii-model-is-validating-but-data-could-not-be-saved

I enabled the log and I have seen this thing:
```
2015-05-08 02:45:53 [-][-][-][info][yii\db\Command::execute] UPDATE `sms_step` SET `smsId`=1, `order`=1, `smsStatusTypeId`=2, `description`='Operacao realizada com sucesso' WHERE `id` IS NULL
```

You can see that the save() method is calling the update() one, why is that if I didn't set the id attribute? If I change save() to insert(), it works.

EDIT:
I am using the TimestampBehavior to set my date field. But I realize that it is setting the $_oldAttributes. What should I do in this case?

Thanks and I am sorry if it is duplicated.

**Comment by unclead** [comment]
I think you should not call `touch` method in this case. This method allow you to update date manually.
Did you configure behavior as described in documentation?
https://github.com/yiisoft/yii2/blob/master/framework/behaviors/TimestampBehavior.php#L39

If yes attributes should be updated automatically without additional calling `touch` method

**Comment by SamMousa** [comment]
@unclead, I think you are right. 

It might be a good improvement for TimestampBehavior to throw an error when used on a new record though.

**Comment by ericmaicon** [comment]
@unclead Thank you for your fast answer. I'll remove the touch call.

**Comment by cebe** [comment]
> It might be a good improvement for TimestampBehavior to throw an error when used on a new record though.

:+1:

**Comment by klimov-paul** [comment]
Resolved by commit cad9fd9
